conans/model/conf.py

```diff
@@ -463,7 +463,7 @@ def compose_conf(self, other):
     def filter_user_modules(self):
         result = Conf()
         for k, v in self._values.items():
-            if _is_profile_module(k):
+            if hasattr(Conf, "_keep_core") or _is_profile_module(k):
```

Comment:
This approach doesn't seem to solve the issue, because it is not the recipe the one with access to ``core``, but the helpers like ``download()`` that get the ``core.xxx`` conf from ``helpers.global_conf``, not from the recipe ``self.conf``